### DISCUSSION TITLE
synopkg and syno_hdd_util commands not found

### DISCUSSION THREAD

**Discussion by macdis** [007revad/Synology_HDD_db#249] [Q&A] [answered]
Running a DS220+ with DSM 7.2.1-69057-4. Recent setup, so not many system customizations. Running the syno_hdd_db.sh script gives the following result:

```
$ sudo -s ./syno_hdd_db.sh -nr
Synology_HDD_db v3.4.84
DS220+ DSM 7.2.1-69057-4
./syno_hdd_db.sh: line 242: synopkg: command not found
Using options: -nr
Running from: /path/to/Synology_HDD_db-3.4.84/syno_hdd_db.sh
./syno_hdd_db.sh: line 772: syno_hdd_util: command not found
./syno_hdd_db.sh: line 772: syno_hdd_util: command not found

ERROR No drives found!
```
Any idea why this is happening? Why these files aren't on root's path?

**Comment by 007revad** [comment]
How is the script running from `/path/to` ? 

Did you create a a root folder named path?

**Comment by macdis** [comment]
I just edited (anonymized) the path info for the purposes of asking the question. The file is in /volume1/homes/EDITED/bin/Synology_HDD_db-3.4.84, and in this case, I ran it from within that directory.

**Reply by 007revad** [reply]
What do the following commands return?
```
which synopkg
which syno_hdd_util
```

**Reply by 007revad** [reply]
You could also try `sudo -E ./syno_hdd_db.sh -nr`

**Comment by macdis** [comment]
Result:
```
synopkg not found
syno_hdd_util not found
```
However they're in `/usr/syno/bin` and can be run using `/usr/syno/bin/synopkg` or  `/usr/syno/bin/syno_hdd_util`.

Also, `sudo -E ./syno_hdd_db.sh -nr` throws the same error messages as `sudo -s`.

**Reply by 007revad** [reply]
What does `echo $PATH` show from a shell as your regular user, and as sudo?

**Reply by macdis** [reply, marked as answer]
Regular administrative user:
```
$ echo $PATH
/var/services/homes/EDITED/bin:/usr/local/bin:/opt/bin:/usr/bin:/bin:/usr/sbin:/sbin
```

Using `sudo -s`, `/usr/syno/bin` is not in $PATH:
```
$ sudo -s echo $PATH
/var/services/homes/EDITED/bin:/usr/local/bin:/opt/bin:/usr/bin:/bin:/usr/sbin:/sbin
```

Using `sudo -i`, `/usr/syno/bin` *is* in $PATH::
```
$ sudo -i
# echo $PATH
/sbin:/bin:/usr/sbin:/usr/bin:/usr/syno/sbin:/usr/syno/bin:/usr/local/sbin:/usr/local/bin
```

That being said..., I just figured out what was going on. Usually, I `ssh` into the NAS using options that allow me to use zsh (and oh-my-zsh) as my shell, without changing it more permanently.
```
RequestTTY yes
RemoteCommand exec /usr/local/bin/zsh -l
```
If I login without those options, which puts me back in `sh`, then $PATH is different and the `syno_hdd_db.sh` script works.

I will update my zsh $PATH variable accordingly.

Thanks for your help!

**Reply by 007revad** [reply]
I just released an updated version with hard-coded /usr/syno/bin/<command> for the all the Synology commands used in the script. 
https://github.com/007revad/Synology_HDD_db/releases/tag/v3.4.85

Do you want to try it to make sure I didn't miss any commands?

**Reply by macdis** [reply]
I'm now getting other errors. Edited output:
```
[...]
./syno_hdd_db.sh: line 208: /usr/syno/bin/get_key_value: No such file or directory
Synology_HDD_db v3.4.85
./syno_hdd_db.sh: line 223: /usr/syno/bin/get_key_value: No such file or directory
./syno_hdd_db.sh: line 224: /usr/syno/bin/get_key_value: No such file or directory
./syno_hdd_db.sh: line 225: /usr/syno/bin/get_key_value: No such file or directory
./syno_hdd_db.sh: line 226: /usr/syno/bin/get_key_value: No such file or directory
[...]
./syno_hdd_db.sh: line 1580: /usr/syno/bin/get_key_value: No such file or directory
./syno_hdd_db.sh: line 1609: /usr/syno/bin/get_key_value: No such file or directory
./syno_hdd_db.sh: line 1785: /usr/syno/bin/get_key_value: No such file or directory
./syno_hdd_db.sh: line 1800: /usr/syno/bin/get_key_value: No such file or directory
[...]
ERROR Failed to disable drive db auto updates!
./syno_hdd_db.sh: line 1832: /usr/syno/bin/get_key_value: No such file or directory
[...]
```

**Reply by macdis** [reply]
These errors seem to be due to the fact that `get_key_value` is in `/usr/bin` rather than `/usr/syno/bin`.

**Reply by 007revad** [reply]
Oops. I should have run the script myself :(

I've changed get_key_value to Synology's synogetkeyvalue (which is in usr/syno/bin) just to keep things consistent. And run it myself.

https://github.com/007revad/Synology_HDD_db/releases/tag/v3.4.86

**Reply by macdis** [reply]
That works, thanks!
